index.js

```diff
@@ -342,31 +342,133 @@
         /**
          * PropTypes
          */
-        
-        function observableTypeChecker (type) {
-            return function(props, propName, componentName) {
-                if (!mobx['isObservable' + type](props[propName])) {
+
+        // Copied from React.PropTypes
+        function createChainableTypeChecker(validate) {
+            function checkType(isRequired, props, propName, componentName, location, propFullName) {
+                componentName = componentName || ANONYMOUS;
+                propFullName = propFullName || propName;
+                if (props[propName] == null) {
+                    if (isRequired) {
+                        var actual = props[propName] === null ? 'null' : 'undefined';
+                        return new Error(
+                          'The ' + location + ' `' + propFullName +'` is marked as required ' +
+                          'in `' + componentName + '`, but its value is `' + actual + '`.'
+                        );
+                    }
+                    return null;
+                } else {
+                    return validate(props, propName, componentName, location, propFullName);
+                }
+            }
+            var chainedCheckType = checkType.bind(null, false);
+            chainedCheckType.isRequired = checkType.bind(null, true);
+            return chainedCheckType;
+        }
+
+        // Copied from React.PropTypes
+        function isSymbol(propType, propValue) {
+            // Native Symbol.
+            if (propType === 'symbol') {
+                return true;
+            }
+
+            // 19.4.3.5 Symbol.prototype[@@toStringTag] === 'Symbol'
+            if (propValue['@@toStringTag'] === 'Symbol') {
+                return true;
+            }
+
+            // Fallback for non-spec compliant Symbols which are polyfilled.
+            if (typeof Symbol === 'function' && propValue instanceof Symbol) {
+                return true;
+            }
+
+            return false;
+        }
+
+        // Copied from React.PropTypes
```

Comment:
No problem, but mind the builds for RN / custom react renderers

Op za 30 jul. 2016 om 20:51 schreef Andy Kogut <notifications@github.com>:

> In index.js
> <https://github.com/mobxjs/mobx-react/pull/93#discussion_r72892525>:
>
> > +            }
> > +
> > +            // 19.4.3.5 Symbol.prototype[@@toStringTag] === 'Symbol'
> > +            if (propValue['@@toStringTag'] === 'Symbol') {
> > +                return true;
> > +            }
> > +
> > +            // Fallback for non-spec compliant Symbols which are polyfilled.
> > +            if (typeof Symbol === 'function' && propValue instanceof Symbol) {
> > +                return true;
> > +            }
> > +
> > +            return false;
> > +        }
> > +
> > +        // Copied from React.PropTypes
>
> Are you ok about building it with browserify? That would enable us to use
> https://github.com/mridgway/hoist-non-react-statics for #92
> <https://github.com/mobxjs/mobx-react/issues/92>
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mobxjs/mobx-react/pull/93/files/38e7bfdb4650eddcb8a256ee47e82649158e304d#r72892525>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABvGhHd-cv7W9PHyKktpFHkPmmZ12ZrCks5qa51OgaJpZM4JY3M7>
> .
>